src/harm.cpp

```diff
@@ -78,6 +79,73 @@ void Harm::AddChild(Object *child)
     Modify();
 }
 
+TransPitch Harm::GetRootPitch(unsigned int &pos)
+{
+    Text *textObject = dynamic_cast<Text *>(this->GetChild(0, TEXT));
+    assert(textObject);
+    std::wstring text = textObject->GetText();
+
+    if (text.at(pos) >= 'A' && text.at(pos) <= 'G') {
+        int pname = (text.at(pos) - 'C' + 7) % 7;
+        int accid = 0;
+        for (pos++; pos < text.length(); pos++) {
+            if (text.at(pos) == L'𝄫')
+                accid -= 2;
+            else if (text.at(pos) == 'b' || text.at(pos) == L'♭')
+                accid--;
+            else if (text.at(pos) == '#' || text.at(pos) == L'♯')
+                accid++;
+            else if (text.at(pos) == L'𝄪')
+                accid += 2;
+            else
+                break;
+        }
+        return TransPitch(pname, accid, 4);
+    }
+    LogWarning("Failed to extract a pitch.");
+    // If we return octave -1, it knows it is an error.
+    return TransPitch(0, 0, -1);
+}
+
+void Harm::SetRootPitch(TransPitch pitch, unsigned int endPos)
+{
+    Text *textObject = dynamic_cast<Text *>(this->GetChild(0, TEXT));
+    assert(textObject);
+    std::wstring text = textObject->GetText();
+
+    textObject->SetText(pitch.GetPitchString() + &text[endPos]);
```

Comment:
Access to character in `text` is not safe. Please change to `.at()` operator and make sure it is not out of range.